### PR TITLE
DEV: Refactor input_to_database.py

### PR DESCRIPTION
**Pull request by thomasyu888**
* abstraction of functions
* add unit tests
* Lint
* Creation of `bin/`

### CODE REVIEW

File: tests/unit/test_input_to_database.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	entity.createdBy='444'
+	#entities = [entity]
+	center = 'SAGE'
+	threads=0
+	testing=False
+
+	fileinfo = {'filePaths':['/path/to/file.csv'],
+				'synId':['syn1234']}
+	with mock.patch.object(syn, "get", return_value=entity) as patch_syn_get,\
+		 mock.patch.object(validate, "validate", return_value=('valid',True)) as patch_validate:
```
Comment (thomasyu888) [inline tests/unit/test_input_to_database.py]: patch to the function.  no object

File: genie/input_to_database.py [hunk 2]
```diff
@@ -1,303 +1,628 @@
...
+        input_error = error_trackerdf[error_trackerdf['id'] == ent.id]
+        if input_validation_status.empty:
+            to_validate = True
         else:
-            statuses.append(checkValid['status'].values[0])
-            if checkError.empty:
-                if checkValid['status'].values[0] == "INVALID":
-                    toValidate =True
+            '''
+            This to_validate is here, because the following is a
```
Comment (kdaily) [inline genie/input_to_database.py]: shouldn't these be comments?

File: genie/input_to_database.py [hunk 3]
```diff
@@ -1,303 +1,628 @@
...
+#         name='Narrow MAF {current_time} Database'.format(
+#             current_time=time.time()),
+#         columns=maf_columns,
+#         parent=process_functions.getDatabaseSynId(
+#             syn, "main",
+#             databaseToSynIdMappingDf=database_synid_mappingdf))
+#     schema.primaryKey = maf_database_ent.primaryKey
+#     new_maf_database = syn.store(schema)
+
+# TODO: Should split this into 3 funcitons
```
Comment (kdaily) [inline genie/input_to_database.py:385]: is there an issue open for this?

File: genie/input_to_database.py [hunk 4]
```diff
@@ -1,303 +1,628 @@
...
-    ################################################################
-    logger.info("GETTING %s INPUT FILES" % center)
-    CLINICAL_PAIR_NAME = ["data_clinical_supp_sample_%s.txt" % center, "data_clinical_supp_patient_%s.txt" % center]
-    walked = synapseutils.walk(syn, synId)
+# Configuration file
+from genie import PROCESS_FILES, process_functions, validate, \
+                  toRetract, write_invalid_reasons
+
+'''
+TODO:
```
Comment (kdaily) [inline genie/input_to_database.py:22]: is there an issue open for this? mention the issue number here.

File: genie/input_to_database.py [hunk 5]
```diff
@@ -374,89 +730,200 @@ def input_to_database(syn, center, process, testing, only_validate, vcf2maf_path
...
+        write_invalid_reasons.write_invalid_reasons(
+            syn, center_mapping_df, error_tracker_synid)
+
 
 if __name__ == "__main__":
-    main()
-    
+    '''
+    Argument parsers
+    TODO: Fix case of arguments
```
Comment (kdaily) [inline genie/input_to_database.py]: open an issue and add issue number.

File: genie/input_to_database.py [hunk 6]
```diff
@@ -374,89 +730,200 @@ def input_to_database(syn, center, process, testing, only_validate, vcf2maf_path
...
+        type=str,
+        help="Path to VEP",
+        default=os.path.expanduser("~/vep"))
+    parser.add_argument(
+        "--vepData",
+        type=str,
+        help="Path to VEP data",
+        default=os.path.expanduser("~/.vep"))
+    '''
+    TODO: Remove thread
```
Comment (kdaily) [inline genie/input_to_database.py]: open an issue and add issue number

**Comment by thomasyu888** [comment]
Fix Batch commands

**Pull request merged**